src/test/groovy/uk/ac/sanger/scgcf/jira/services/actions/CherrypickActionTest.groovy

```diff
@@ -0,0 +1,99 @@
+package uk.ac.sanger.scgcf.jira.services.actions
+
+import spock.lang.Specification
+import uk.ac.sanger.scgcf.jira.services.models.*
+
+/**
+ * Created by rf9 on 27/06/2016.
+ */
+class CherrypickActionTest extends Specification {
+
+    def "should cherrypick from one plate to another"() {
+        given:
+        def labwareType = new LabwareType(name: 'test_type', layout: new Layout(name: 'four plate layout'))
+        def locations = [new Location(name: 'A1'), new Location(name: 'A2'), new Location(name: 'B1'), new Location(name: 'B2')]
+        def sourceMaterials = (1..4).collect { new Material(id: "source_id_$it", name: "source_material_$it") }
+        def materialType = new MaterialType(name: 'new type')
+
+        def sourceLabware = new Labware(barcode: 'TEST_001', labwareType: labwareType, receptacles: (0..3).collect { new Receptacle(materialUuid: sourceMaterials[it].id, location: locations[it]) })
+        def destinationLabware = new Labware(barcode: 'TEST_002', labwareType: labwareType, receptacles: locations.collect { new Receptacle(location: it) })
+
+        def newMaterials = []
+        GroovySpy(TransferActions, global: true)
+
+        def transferMap = [
+            new TransferMapping(sourceBarcode: 'TEST_001', destinationBarcode: 'TEST_002', sourceLocation: 'A1', destinationLocation: 'A2'),
+            new TransferMapping(sourceBarcode: 'TEST_001', destinationBarcode: 'TEST_002', sourceLocation: 'B1', destinationLocation: 'B2')
+        ]
+
+        when:
+        TransferActions.cherrypick([sourceLabware], [destinationLabware], materialType, transferMap)
+
+        then:
+        1 * TransferActions.getMaterialsByUuid([sourceMaterials[0].id, sourceMaterials[2].id]) >> [sourceMaterials[0], sourceMaterials[2]]
+        1 * TransferActions.postNewMaterials(_) >> { materials ->
+            newMaterials += materials[0].eachWithIndex { material, i ->
+                material.id = "${material.name}_uuid"
+            }
+        }
+        1 * TransferActions.updateLabware(destinationLabware) >> destinationLabware
```

Comment:
Because TransferActions.updateLabware is private - which is fine - we should change it to:
1 * Labware.update() >> destinationLabware
or something similar in every related tests